tests/integration/interactive/cmd_init_test.go

```diff
@@ -0,0 +1,42 @@
+package interactive
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-developer/odo/tests/helper"
+)
+
+var _ = Describe("odo login and logout command tests", func() {
+
+	var commonVar helper.CommonVar
+	var interVar helper.Interactive
+
+	// This is run before every Spec (It)
+	var _ = BeforeEach(func() {
+		commonVar = helper.CommonBeforeEach()
+	})
+
+	// // Clean up after the test
+	// // This is run after every Spec (It)
+	var _ = AfterEach(func() {
+		helper.CommonAfterEach(commonVar)
+	})
+	It("should download correct devfile", func() {
+		interVar.Command = []string{"odo", "init"}
+		interVar.ExpectFromtty = append(interVar.ExpectFromtty, "Select language")
+		interVar.SendOntty = append(interVar.SendOntty, "go")
+		interVar.ExpectFromtty = append(interVar.ExpectFromtty, "Select project type")
+		interVar.SendOntty = append(interVar.SendOntty, "\n")
+		interVar.ExpectFromtty = append(interVar.ExpectFromtty, "Which starter project do you want to use")
+		interVar.SendOntty = append(interVar.SendOntty, "\n")
+		interVar.ExpectFromtty = append(interVar.ExpectFromtty, "Enter component name")
+		interVar.SendOntty = append(interVar.SendOntty, "my-go-app")
+		interVar.ExpectFromtty = append(interVar.ExpectFromtty, "Your new component \"my-go-app\" is ready in the current directory.")
```

Comment:
I'm not very comfortable with this way to do, as we are very limited in what we could do: only expecting and sending.

I think it would be nice to pass a function to the RunInteractive command, and that function would be executed from inside the RunInteractive one, after the initialization is done. So we would be free to do anything we want into this passed function.

Reply:
> I think it would be nice to pass a function to the RunInteractive command, and that function would be executed from inside the RunInteractive one, after the initialization is done. So we would be free to do anything we want into this passed function.

@feloy can you help me understand this better?